src/components/search/data/hooks.js

```diff
@@ -107,19 +108,48 @@ export const useNbHitsFromSearchResults = (searchResults) => {
   return nbHits;
 };
 
-export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan }) => {
+export const getCatalogString = (offerCatalogs, initialString = '') => {
+  const offerCatalogsCopy = [...offerCatalogs];
+  const lastCatalog = offerCatalogsCopy.pop();
+  return `${offerCatalogsCopy.reduce(
+    (result, catalog) => `${result}enterprise_catalog_uuids:${catalog} OR `, initialString,
+  )}enterprise_catalog_uuids:${lastCatalog}`;
```

Comment:
It feels a bit odd to need to pop off the last catalog in the list. Just a suggestion/opinion, but I find this to be a bit easier to read/interpret:

```jsx
export const getCatalogString = (catalogs) => {
  function catalogFilterReducer(result, catalog, index) {
    const isLastCatalog = index === catalogs.length - 1;
    let query = `${result}enterprise_catalog_uuids:${catalog}`;
    if (!isLastCatalog) {
      query += ' OR '; 
    }
    return query;
  }
  
  return catalogs.reduce(catalogFilterReducer, '');
};
```